pretty_midi/instrument.py

```diff
@@ -205,6 +205,93 @@ def get_end_time(self):
+                                  normalize=False):
+        """Computes the frequency of pitch classes of the current instrument,
+        optionally weighted by their durations or velocities.
+
+        Parameters
+        ----------
+        use_duration : bool
+            Weight frequency by note duration
+        use_velocity : bool
+            Weight frequency by note velocity
+        normalize : bool
+            Normalizes the histogram such that the sum of bin values is 1.
+
+        Returns
+        -------
+        histogram : np.ndarray, shape=(12,)
+            Histogram of pitch classes given current instrument, optionally
+            weighted by their durations or velocities
+        """
+
+        # Return all zeros if track is drum
+        if self.is_drum:
+            return np.zeros(12)
+
+        weights = np.ones(len(self.notes))
+
+        # Assumes that duration and velocity have equal weight
+        if use_duration:
+            weights *= [note.end - note.start for note in self.notes]
+        if use_velocity:
+            weights *= [note.velocity for note in self.notes]
+
+        histogram, _ = np.histogram([n.pitch % 12 for n in self.notes],
+                                    bins=np.arange(13),
+                                    weights=weights,
+                                    density=normalize)
+
+        return histogram
+
+    def get_pitch_class_transition_matrix(self, normalize=False):
+        """Computes the pitch class transition matrix of the current instrument
+
+        Transitions are added whenever the end of a note is within 50ms from the
+        start of any other note.
+
+        Parameters
+        ----------
+        normalize : bool
+            Normalize transition matrix such that matrix sum equals is 1.
+
+        Returns
+        -------
+        pitch_class_transition_matrix : np.ndarray, shape=(12,12)
+            Pitch class transition matrix
+        """
+
+        # instrument is drum or less than one note, return all zeros
+        if self.is_drum or len(self.notes) <= 1:
+            return np.zeros((12, 12))
+
+        # container for holding pitch class transitions
+        pitch_class_transition_matrix = np.zeros((12, 12))
+
+        # use 20hz(0.05s) as the minimun time threshold for separate notes
+        time_thresh = 0.05
+
+        # data matrix
+        data_mat = np.array([[x.start, x.end, x.pitch % 12]
+                             for x in self.notes])
+
+        # compute distance matrix for all end and start time pairs
+        dist_mat = np.subtract.outer(data_mat[:, 1], data_mat[:, 0])
+
+        # compute boolean matrix of indices within threshold
+        trans_idx= abs(dist_mat) < time_thresh
+
+        # update transition matrix
+        for i in xrange(len(data_mat)):
+            pitch_class_transition_matrix[
+                data_mat[i, 2], data_mat[trans_idx[i], 2].astype(int)] += 1
```

Comment:
Ah, they're just referring to the fact that you shouldn't use it directly to index an array (which we aren't doing).  In this use-case, of course it's suitable, because it works!